errors/errors.js

```diff
@@ -7,8 +7,10 @@ exports.handleCustomErrors = (err, request, response, next) => {
 };
 
 exports.handle400Errors = (err, request, response, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02"||err.code === "23502") {
     response.status(400).send({ msg: "Bad request" });
+  } else if (err.code === "23503") {
+    response.status(404).send({ msg: "Username doesn't exist" });
```

Comment:
This error code means a foreign key violation so it can be risky using this with a custom error message. You run the risk of throwing this error if a different foreign key violation occurs! I would instead check the user exists a different way, may use SQL to query the users table in the model 😉 

Also, this is a 404 test but you have put it in a 400 handler!